### PR TITLE
Align components

### PR DESCRIPTION
**Pull request by sz332**
### PR Checklist
Please check if your PR fulfills the following requirements:

- [X] I have read, and I am following the [Contributor guide](https://github.com/gaphor/gaphor/blob/main/CONTRIBUTING.md)
- [X] I have read, and I understand the GNOME [Code of Conduct](https://wiki.gnome.org/Foundation/CodeOfConduct)

### PR Type
What kind of change does this PR introduce?

- [ ] Bug fix
- [X] Feature
- [ ] Chore (refactoring, formatting, local variables, other cleanup)
- [ ] Documentation content changes

### What is the current behavior?
Currently the user cannot align multiple elements to the left, bottom, top, right, etc.

Issue Number: #1476 

### What is the new behavior?

Using the menu / tools now the user will be able to align components.

### Does this PR introduce a breaking change?
- [ ] Yes
- [X] No

### CODE REVIEW

**Comment by danyeaw** [comment]
Hey @sz332, I like this idea to add these to the Tools menu as a first MVP!

I tried to do an alignment and I got the following traceback:

```
gaphor.transaction ERROR Transaction terminated due to an exception, performing a rollback
gaphor.plugins.errorreports.errorreports ERROR Uncaught exception
Traceback (most recent call last):
  File "/home/dan/Projects/gaphor/gaphor/ui/actiongroup.py", line 145, in _action_activate
    method()
  File "/home/dan/Projects/gaphor/gaphor/plugins/align/align.py", line 36, in align_left
    self._modify_elements(self._align_elements_left)
  File "/home/dan/Projects/gaphor/gaphor/plugins/align/align.py", line 81, in _modify_elements
    f(elements)
  File "/home/dan/Projects/gaphor/gaphor/plugins/align/align.py", line 85, in _align_elements_left
    left_edge = self._left_edge(elements)
                ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/dan/Projects/gaphor/gaphor/plugins/align/align.py", line 145, in _left_edge
    return min({self._pos_x(item) for item in elements})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: min() arg is an empty sequence
```

**Comment by sz332** [comment]
> Hey @sz332, I like this idea to add these to the Tools menu as a first MVP!
> 
> I tried to do an alignment and I got the following traceback:
> 
> ```
> gaphor.transaction ERROR Transaction terminated due to an exception, performing a rollback
> gaphor.plugins.errorreports.errorreports ERROR Uncaught exception
> Traceback (most recent call last):
>   File "/home/dan/Projects/gaphor/gaphor/ui/actiongroup.py", line 145, in _action_activate
>     method()
>   File "/home/dan/Projects/gaphor/gaphor/plugins/align/align.py", line 36, in align_left
>     self._modify_elements(self._align_elements_left)
>   File "/home/dan/Projects/gaphor/gaphor/plugins/align/align.py", line 81, in _modify_elements
>     f(elements)
>   File "/home/dan/Projects/gaphor/gaphor/plugins/align/align.py", line 85, in _align_elements_left
>     left_edge = self._left_edge(elements)
>                 ^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/home/dan/Projects/gaphor/gaphor/plugins/align/align.py", line 145, in _left_edge
>     return min({self._pos_x(item) for item in elements})
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> ValueError: min() arg is an empty sequence
> ```

It is fixed with the last commit, can you please re-check?

File: gaphor/plugins/align/align.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+from gaphor.ui.event import DiagramSelectionChanged
+from gaphor.core import action, event_handler
+from gaphor.diagram.presentation import ElementPresentation
+
+from gaphor.transaction import Transaction
+
+
+class AlignService(Service, ActionProvider):
+    def __init__(self, event_manager, diagrams, tools_menu=None, dump_gv=False):
```
Comment (danyeaw) [inline gaphor/plugins/align/align.py:14]: I know that the pydot service uses the dump_gv parameter, is it needed for this service?

**Review by danyeaw — APPROVED** [review]
Hi @sz332, this looks really great - I can't believe what a nice feature this is in such a small amount of code :1st_place_medal:

File: gaphor/plugins/align/align.py (hunk 1 above)
Comment (sz332) [reply gaphor/plugins/align/align.py]: Good catch, removed the unnecessary parts.

**Pull request merged**